src/components/Avatar/Avatar.tsx

```diff
@@ -40,46 +47,52 @@ export const Avatar = forwardRef<HTMLSpanElement, AvatarProps>(function Avatar(
     size,
     style = {},
     onError,
+    onClick,
     ...props
   },
   ref,
 ) {
   const hash = useIdColorHash(id);
   const fallbackInitial = <>{getInitialLetter(name)}</>;
 
-  return (
-    <span
-      ref={ref}
-      role="img"
-      title={id}
-      {...props}
-      aria-label=""
-      data-type={type}
-      data-color={hash}
-      className={classnames(styles.avatar, className)}
-      style={
-        {
-          ...style,
-          "--cpd-avatar-size": size,
-        } as React.CSSProperties
-      }
-    >
-      {!src ? (
-        fallbackInitial
-      ) : (
-        <Suspense fallback={fallbackInitial}>
-          <SuspenseImg
-            src={src}
-            className={classnames(styles.image)}
-            data-type={type}
-            style={style}
-            width={size}
-            height={size}
-            title={id}
-            onError={onError}
-          />
-        </Suspense>
-      )}
-    </span>
+  return React.createElement(
+    // Preventing `span` elements to have an `onClick` as it is usually
+    // terrible for accesibility
```

Comment:
Always* as keyboard-only users can't click on things, so you'd need to also handle onKeyDown & onKeyUp